lib/charms/mysql/v0/tls.py

```diff
@@ -174,6 +174,9 @@ def _on_tls_relation_broken(self, _) -> None:
         except KeyError:
             # ignore key error for unit teardown
             pass
+        if self.charm.removing_unit:
```

Comment:
im wondering if it might be a good idea to add this check to all event handlers (or better yet, condition all `self.framework.observe` statements to ensure no event handlers run if charm is being removed)

to better understand: why are we allowing setting of secrets on lines 170-173 instead of just no-oping the handler entirely

Reply:
yeah, eventually.

Reply:
> to better understand: why are we allowing setting of secrets on lines 170-173 instead of just no-oping the handler entirely

indeed. If unit is going away, so will it's databag keys